tests/math/test_density_matrices.py

```diff
@@ -194,3 +194,136 @@ def test_wrong_shape_jax_jit(self):
+            density_matrix, indices=wires
+        )
+        assert np.allclose(density_matrix, returned_density_matrix)
+
+    @pytest.mark.parametrize("density_matrix, expected_density_matrix", density_matrices)
+    @pytest.mark.parametrize("wires", single_wires_list)
+    def test_to_density_matrix_with_matrix_single_wires(
+        self, density_matrix, wires, expected_density_matrix
+    ):
+        """Test the to_density_matrix with matrix for single wires."""
+        density_matrix = fn.to_density_matrix(density_matrix, indices=wires)
+        assert np.allclose(density_matrix, expected_density_matrix[wires[0]])
+
+    @pytest.mark.parametrize("density_matrix, expected_density_matrix", density_matrices)
+    @pytest.mark.parametrize("wires", multiple_wires_list)
+    def test_to_density_matrix_with_matrix_full_wires(
+        self, density_matrix, wires, expected_density_matrix
+    ):
+        """Test the to_density_matrix with matrix for full wires."""
+        returned_density_matrix = fn.to_density_matrix(density_matrix, indices=wires)
+        assert np.allclose(density_matrix, returned_density_matrix)
+
+    @pytest.mark.parametrize("density_matrix, expected_density_matrix", density_matrices)
+    @pytest.mark.parametrize("wires", multiple_wires_list)
+    def test_density_matrix_from_matrix_check(self, density_matrix, wires, expected_density_matrix):
+        """Test the density matrix from matrices for single wires with state checking"""
+        returned_density_matrix = fn.quantum._density_matrix_from_matrix(
+            density_matrix, indices=wires, check_state=True
+        )
+        assert np.allclose(density_matrix, returned_density_matrix)
+
+    def test_matrix_wrong_shape(self):
+        """Test that wrong shaped state vector raises an error with check_state=True"""
+        density_matrix = [[1, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+        with pytest.raises(ValueError, match="Density matrix must be of shape"):
+            fn.quantum._density_matrix_from_matrix(density_matrix, indices=[0], check_state=True)
+
+    def test_matrix_wrong_trace(self):
+        """Test that density matrix with wrong trace raises an error with check_state=True"""
+        density_matrix = [[0.1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]
+
+        with pytest.raises(ValueError, match="The trace of the density matrix should be one."):
+            fn.quantum._density_matrix_from_matrix(density_matrix, indices=[0], check_state=True)
+
+    def test_matrix_not_hermitian(self):
+        """Test that non hermitian matrix raises an error with check_state=True"""
+        density_matrix = [[0.1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0.5, 0.9]]
+
+        with pytest.raises(ValueError, match="The matrix is not hermitian."):
+            fn.quantum._density_matrix_from_matrix(density_matrix, indices=[0], check_state=True)
+
+    def test_density_matrix_from_state_vector_jax_jit(self):
```

Comment:
Since we're testing jitting with JAX, it may be worth testing `tf.function` and autograph mode [as here](https://github.com/PennyLaneAI/pennylane/blob/master/tests/interfaces/test_tensorflow_qnode.py#L1336).